backend/src/v1/case_file/case_file.service.ts

```diff
@@ -234,6 +234,26 @@ export class CaseFileService {
     ${caseFileQueryFields}
     }`;
 
+    // If the assessment being updated has no action required or the justification is not duplicate, then check if there
+    // was previously a link to another complaint. If there is, remove it.
+    if (
+      modelAsAny.updateAssessmentInput.assessmentDetails.actionNotRequired ||
+      modelAsAny.updateAssessmentInput.assessmentDetails.actionJustificationCode !== "DUPLICATE"
+    ) {
```

Comment:
This works in the intended cases, but it also catches changes to the date or the officer when the link itself isn't changing. We'll need to check what the values are currently against what they're changing to here, so we could potentially pass this function the existing outcome from the frontend to compare against without having to fetch it here.

Reply:
It seems to be creating a new link and archiving the old link which shouldn't be a big deal. We end up with the correct information and I don't think this is a common scenario.